pkg/kubelet/kubelet.go

```diff
@@ -1311,7 +1311,14 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if errOuter := canRunPod(pod); errOuter != nil || pod.DeletionTimestamp != nil || apiPodStatus.Phase == api.PodFailed {
+	errOuter := canRunPod(pod)
+	if errOuter == nil {
+		rs := kl.runtimeState.networkErrors()
```

Comment:
Oh I think I understand...

Perhaps if we renamed to `networkPluginErrors`, or `podNetworkErrors`?  The idea is that if the network plugin has a problem, we put it into this separate list.  An error that prevents hostNetwork pods would presumably not be a network plugin error, and would go into the general errors list (which prevents the pod sync loop from even starting)